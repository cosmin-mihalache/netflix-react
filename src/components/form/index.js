import React from 'react';
import {
  Container,
  Base,
  Title,
  Submit,
  Text,
  SmallText,
  Input,
  Link,
  Error,
} from './styles/form';

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Input = function FormInput({ children, ...restProps }) {
  return <Input {...restProps}>{children}</Input>;
};

Form.Submit = function FormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Link = function FormLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Form.Text = function FormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Form.SmallText = function FormSmallText({ children, ...restProps }) {
  return <SmallText {...restProps}>{children}</SmallText>;
};
