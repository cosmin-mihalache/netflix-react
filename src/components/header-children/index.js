import React from 'react';
import { Container, Title, SubTitle } from './styles/header-children';

export default function HeaderChild({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

HeaderChild.Title = function HeaderChildTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

HeaderChild.SubTitle = function HeaderChildSubTitle({
  children,
  ...restProps
}) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};
