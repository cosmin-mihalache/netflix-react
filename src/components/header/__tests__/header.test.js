import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { Header } from '../../../components';

jest.mock('react-router-dom');

describe('<Header />', () => {
  it('renders the basic <Header /> with a background', () => {
    const { container, getByText, getByTestId } = render(
      <Header>
        <Header.Frame>
          <Header.Logo src="/images/misc/logo.svg" alt="Netflix" />
          <Header.TextLink active="true">Hello, I'm a link!</Header.TextLink>
        </Header.Frame>
      </Header>
    );

    expect(getByText("Hello, I'm a link!")).toBeTruthy();
    expect(getByTestId('header-bg')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the basic <Header /> without a background', () => {
    const { container, getByText, queryByTestId } = render(
      <Header bg={false}>
        <Header.Frame>
          <Header.Logo src="/images/misc/logo.svg" alt="Netflix" />
          <Header.ButtonLink>Sign In</Header.ButtonLink>
          <Header.TextLink active="false">Hello, I'm a link!</Header.TextLink>
        </Header.Frame>
      </Header>
    );

    expect(getByText("Hello, I'm a link!")).toBeTruthy();
    expect(queryByTestId('header-bg')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the full <Header /> with a background', () => {
    const { container, getByText, getByTestId } = render(
      <Header src="joker" dontShowOnSmallViewPort>
        <Header.Frame>
          <Header.Group>
            <Header.Logo src="/images/misc/logo.svg" alt="Netflix" />
            <Header.TextLink active={false} onClick={() => {}}>
              Series
            </Header.TextLink>
            <Header.TextLink active onClick={() => {}}>
              Films
            </Header.TextLink>
          </Header.Group>
          <Header.Group>
            <Header.Search searchTerm="Joker" setSearchTerm={() => {}} />
            <Header.Profile>
              <Header.Picture src="/images/cosmin.png" />
              <Header.Dropdown>
                <Header.Group>
                  <Header.Picture src="/images/cosmin.png" />
                  <Header.TextLink>Cosmin Mihalache</Header.TextLink>
                </Header.Group>
                <Header.Group>
                  <Header.TextLink onClick={() => {}}>Sign Out</Header.TextLink>
                </Header.Group>
              </Header.Dropdown>
            </Header.Profile>
          </Header.Group>
        </Header.Frame>
        <Header.Feature>
          <Header.FeatureCallOut>Watch Joker now</Header.FeatureCallOut>
          <Header.Text>Forever alone in a crowd...</Header.Text>
          <Header.PlayButton>Play</Header.PlayButton>
        </Header.Feature>
      </Header>
    );

    expect(getByTestId('search-input')).toBeTruthy();
    expect(getByTestId('search-input').value).toBe('Joker');
    fireEvent.change(getByTestId('search-input'), {
      target: { value: 'Simpsons' },
    });
    fireEvent.click(getByTestId('search-click'));

    expect(getByText('Series')).toBeTruthy();
    expect(getByText('Films')).toBeTruthy();
    expect(getByText('Cosmin Mihalache')).toBeTruthy();
    expect(getByText('Watch Joker now')).toBeTruthy();
    expect(getByText('Sign Out')).toBeTruthy();
    expect(getByText('Forever alone in a crowd...')).toBeTruthy();
    expect(getByText('Play')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
