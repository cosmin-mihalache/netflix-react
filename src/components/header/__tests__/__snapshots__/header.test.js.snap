// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the basic <Header /> with a background 1`] = `
<section
  class="header__Background-sc-1b57ikp-0 ALWkT"
  data-testid="header-bg"
>
  <div
    class="header__Container-sc-1b57ikp-2 hvyReA"
  >
    <p
      class="header__Link-sc-1b57ikp-4 foLSrX"
    >
      Hello, I'm a link!
    </p>
  </div>
</section>
`;

exports[`<Header /> renders the basic <Header /> without a background 1`] = `
<div
  class="header__Container-sc-1b57ikp-2 hvyReA"
>
  <p
    class="header__Link-sc-1b57ikp-4 ittReb"
  >
    Hello, I'm a link!
  </p>
</div>
`;

exports[`<Header /> renders the full <Header /> with a background 1`] = `
<section
  class="header__Background-sc-1b57ikp-0 eZbbUT"
  data-testid="header-bg"
  src="joker"
>
  <div
    class="header__Container-sc-1b57ikp-2 hvyReA"
  >
    <div
      class="header__Group-sc-1b57ikp-1 fCiitE"
    >
      <p
        class="header__Link-sc-1b57ikp-4 ittReb"
      >
        Series
      </p>
      <p
        class="header__Link-sc-1b57ikp-4 ittReb"
      >
        Films
      </p>
    </div>
    <div
      class="header__Group-sc-1b57ikp-1 fCiitE"
    >
      <div
        class="header__Search-sc-1b57ikp-6 fjGAMt"
      >
        <button
          class="header__SearchIcon-sc-1b57ikp-7 hBxdhO"
          data-testid="search-click"
        >
          <img
            alt="Search"
            src="/images/icons/search.png"
          />
        </button>
        <input
          class="header__SearchInput-sc-1b57ikp-9 eRLizJ"
          data-testid="search-input"
          placeholder="Search films and series"
          value="Joker"
        />
      </div>
      <div
        class="header__Profile-sc-1b57ikp-10 fUkepP"
      >
        <button
          class="header__Picture-sc-1b57ikp-3 enshzs"
          src="/images/users//images/cosmin.png.png"
        />
        <div
          class="header__Dropdown-sc-1b57ikp-5 kYqXN"
        >
          <div
            class="header__Group-sc-1b57ikp-1 fCiitE"
          >
            <button
              class="header__Picture-sc-1b57ikp-3 enshzs"
              src="/images/users//images/cosmin.png.png"
            />
            <p
              class="header__Link-sc-1b57ikp-4 ittReb"
            >
              Cosmin Mihalache
            </p>
          </div>
          <div
            class="header__Group-sc-1b57ikp-1 fCiitE"
          >
            <p
              class="header__Link-sc-1b57ikp-4 ittReb"
            >
              Sign Out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-sc-1b57ikp-2 header__Feature-sc-1b57ikp-13 fhjVyT"
  >
    <h2
      class="header__FeatureCallOut-sc-1b57ikp-15 gAIdxB"
    >
      Watch Joker now
    </h2>
    <p
      class="header__Text-sc-1b57ikp-14 hIdodF"
    >
      Forever alone in a crowd...
    </p>
    <button
      class="header__PlayButton-sc-1b57ikp-8 cldigT"
    >
      Play
    </button>
  </div>
</section>
`;
